core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java

```diff
@@ -108,21 +115,47 @@ private void registerEth2Routes(
 
   private ArtifactSignerProvider loadSigners(
       final Config config, final Vertx vertx, final MetricsSystem metricsSystem) {
-    final HashicorpConnectionFactory hashicorpConnectionFactory =
-        new HashicorpConnectionFactory(vertx);
-
-    final AbstractArtifactSignerFactory artifactSignerFactory =
-        new BlsArtifactSignerFactory(
-            config.getKeyConfigPath(),
-            metricsSystem,
-            hashicorpConnectionFactory,
-            BlsArtifactSigner::new);
-
-    final Collection<ArtifactSigner> signers =
-        SignerLoader.load(
-            config.getKeyConfigPath(),
-            "yaml",
-            new YamlSignerParser(List.of(artifactSignerFactory)));
+
+    final List<ArtifactSigner> signers = Lists.newArrayList();
+    if (config.getKeyConfigPath() != null) {
+      final HashicorpConnectionFactory hashicorpConnectionFactory =
+          new HashicorpConnectionFactory(vertx);
+
+      final AbstractArtifactSignerFactory artifactSignerFactory =
+          new BlsArtifactSignerFactory(
+              config.getKeyConfigPath(),
+              metricsSystem,
+              hashicorpConnectionFactory,
+              BlsArtifactSigner::new);
+
+      signers.addAll(
+          SignerLoader.load(
+              config.getKeyConfigPath(),
+              "yaml",
+              new YamlSignerParser(List.of(artifactSignerFactory))));
+    }
+
+    if (config.getAzureKeyVaultParameters() != null) {
+      final AzureKeyVaultParameters params = config.getAzureKeyVaultParameters();
+      final AzureKeyVault keyVault =
+          new AzureKeyVault(
+              params.getClientlId(),
+              params.getClientSecret(),
+              params.getTenantId(),
+              params.getKeyVaultName());
+      signers.addAll(
+          keyVault
+              .getAvailableSecrets()
+              .parallelStream()
+              .map(
+                  secret -> {
+                    final Bytes privateKeyBytes = Bytes.fromHexString(secret);
```

Comment:
should we wrap it in a try/catch block and log if "conversion" of specific secret fails instead of error-ing out completely?

Reply:
done